### ISSUE TITLE
detection of Artifactory NuGet V3 feed is wrong

### ISSUE THREAD

**Issue by tomaxsas** [fsprojects/Paket#3351]
### Description

Paket detects v3 feed of artifactory in wrong way. feed is on `http://<host>/<path>/api/nuget/v3/<repo_name>` there is no index.json at the end
wrong source code line:
https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Versioning/PackageSources.fs#L153

source
https://www.jfrog.com/confluence/display/RTF/NuGet+Repositories#NuGetRepositories-ConfigureNuGetCLI/VisualStudiotoWorkwithNuGetv3API
https://jfrog.com/knowledge-base/why-artifactory-is-working-with-nuget-v2-protocol-and-not-nuget-v3-while-the-client-is-sending-api-v3-requests/

### Repro steps

1. Add source in from like http://<host>/<path>/api/nuget/v3/<repo_name>

2. Try to restore nuget packages

### Expected behavior

Correct feed detection or forcing feed version from config

### Actual behavior

v3 feed detected as v2

### Known workarounds

None

**Comment by matthid** [comment]
It seems like those Artifactory guys really want to do things differently every time :/

**Comment by viktor-svub** [comment]
You are right -- GET on `index.json` returns 405, and GET on the v3 feed returns the json content; It's like Artifactory is getting difficult by design :( 

What is strange, it _seemed_ to work for a while, and even according to the bottom picture on the aforementioned docs page: 
[](https://jfrog.com/knowledge-base/why-artifactory-is-working-with-nuget-v2-protocol-and-not-nuget-v3-while-the-client-is-sending-api-v3-requests/, the `index.json` should-be/have-been part of the "NuGet v3 feed URL". 

With this, convention-based detection is getting unmaintainable - I was already experimenting with header-based server & protocol detection, and will return to that as soon as time allows.

**Comment by matthid** [comment]
At this time we could also assume v3 if we cannot detect v2, I don't think that would be breaking because we have the v3->v2 fallback in place

**Comment by enricosada** [comment]
I'll fix that

**Comment by gantrior** [comment]
@enricosada do you still plan to fix that?
